src/vtp/script_libs/cast_ballot_lib.py

```diff
@@ -0,0 +1,334 @@
+#!/usr/bin/env python
+
+#  VoteTrackerPlus
+#   Copyright (C) 2022 Sandy Currier
+#
+#   This program is free software; you can redistribute it and/or modify
+#   it under the terms of the GNU General Public License as published by
+#   the Free Software Foundation; either version 2 of the License, or
+#   (at your option) any later version.
+#
+#   This program is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU General Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License along
+#   with this program; if not, write to the Free Software Foundation, Inc.,
+#   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+"""
+LIbrary backend to command line level test script to automatically cast a ballot.
+
+See 'cast_ballot.py -h' for usage information.
+"""
+
+# pylint: disable=wrong-import-position
+import argparse
+import logging
+import os
+import pprint
+import random
+import sys
+import traceback
+
+import pyinputplus
+
+# Local imports
+from vtp.utils.address import Address
+from vtp.utils.ballot import Ballot, BlankBallot, Contests
+from vtp.utils.common import Globals, Shellout
+
+
+class CastBallotLib:
+    """A class to wrap the cast_ballot.py script."""
+
+    def __init__(self, argv):
+        """Only to module-ize the scripts and keep things simple and idiomatic."""
+        self.argv = argv
+        self.parsed_args = None
+        self.parse_arguments()
+
+    def __str__(self):
+        """Boilerplate"""
+        return "argv=" + str(self.argv) + "\n" + "parsed_args=" + str(self.parsed_args)
+
+    def make_random_selection(self, the_ballot, the_contest):
+        """Will randomly make selections on a contest"""
+        # get the possible choices
+        choices = the_contest.get("choices")
+        tally = the_contest.get("tally")
+        # choose something randomly
+        picks = list(range(len(choices)))
+        # For plurality and max=1, the first choice is the only
+        # choice.  For plurality and max>1, the order does not matter
+        # - a selection is a selection.  For RCV, the order does
+        # matter as that is the ranking.
+        random.shuffle(picks)
+        if "plurality" == tally:
+            loop = the_contest.get("max")
+        elif "rcv" == tally:
+            loop = len(choices)
+        else:
+            raise KeyError(f"Unspoorted tally ({tally})")
+        while loop > 0:
+            the_ballot.add_selection(the_contest, picks.pop(0))
+            loop -= 1
+
+    def get_user_selection(self, the_ballot, the_contest, count, total_contests):
```

Comment:
```suggestion
    def get_user_selection(self, ballot, contest, count, total_contests):
```
Now that this isn't global the `the_` prefixes aren't necessary.

Reply:
Actually, I was arbitrarily adding a 'the_' prefix - I would have been equally happy using a 'a_' prefix.  I just wanted something other than 'ballot' and 'contest' so that tag searching/query/replace would not hit on every text instance of 'ballot' or 'contest'.

Can you suggest something else beside 'the_' or 'a_' prefix?  I don't see this as a big deal that is worth changing, but am happy to tag replace it with something else if you suggest something.  ?